src/kernel/page.h

```diff
@@ -0,0 +1,44 @@
+/* Though page flags are just a u64, we hide it behind this type to
+   emphasize that page flags should be composed using helpers with
+   clear semantics, not architecture bits. This is to avoid mistakes
+   due to a union of PAGE_* constants on one architecture meaning
+   something entirely different on another. */
+
+typedef struct pageflags {
+    u64 w;                      /* _PAGE_* flags, keep private to page.[hc] */
```

Comment:
Could remove the initial underscore from `_PAGE_*` in the comment, since these constants have been renamed without the underscore.